packages/query-core/src/types.ts

```diff
@@ -544,7 +549,12 @@ export type MutationKey = readonly unknown[]
 
 export type MutationStatus = 'idle' | 'loading' | 'success' | 'error'
 
-export interface MutationMeta {
+export interface MutationMeta<
+  TData = unknown,
+  TError = unknown,
+  TVariables = void,
```

Comment:
I don't think this should be void. It's c/p from `MutationOptions`, and I think it's typed as `unknown` in `MutationCacheConfig` - so my example will fail with `Argument of type 'unknown' is not assignable to parameter of type 'void'`
```suggestion
  TVariables = unknown,
```